src/lib/rest/rest.cpp

```diff
@@ -614,6 +614,65 @@ bool urlCheck(ConnectionInfo* ciP, const std::string& url)
+   * a substring of other, we search first for the longest ones
+   *
+   *  updateContextAvailabilitySubscription
+   *  unsubscribeContextAvailability
+   *  subscribeContextAvailability
+   *  discoverContextAvailability
+   *  updateContextSubscription
+   *  unsubscribeContext
+   *  subscribeContext
+   *  registerContext
+   *  updateContext
+   *  queryContext
+   *
+   * Note that in the case of unsubscribeContext and unsubscribeContextAvailability doesn't
+   * actually use a servicePath, but we also provide a default in that case for the sake of
+   * completeness
+   */
+
+  // FIXME P5: this strategy can be improved taking into account full URL. Otherwise, it is
+  // ambigous (e.g. entity which id is "queryContext" and are using a conv op). However, it is
+  // highly improbable that the user uses entities which id (or type) match the name of a
+  // standard operation
+
+  // FIXME P5: unhardwire literals
+  if      (strcasestr(url, "updateContextAvailabilitySubscription") != NULL) return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "unsubscribeContextAvailability") != NULL)        return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "subscribeContextAvailability") != NULL)          return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "discoverContextAvailability") != NULL)           return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "updateContextSubscription") != NULL)             return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "unsubscribeContext") != NULL)                    return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "subscribeContext") != NULL)                      return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "registerContext") != NULL)                       return DEFAULT_SERVICE_PATH;
+  else if (strcasestr(url, "updateContext") != NULL)                         return DEFAULT_SERVICE_PATH;
+  else if (strcasestr(url, "queryContext") != NULL)                          return DEFAULT_SERVICE_PATH_RECURSIVE;
+
+  /* Look for convenience operation. Subscription-related operations are special, all the other depend on
+   * the method
+   */
+  else if (strcasestr(url, "contextAvailabilitySubscriptions") != NULL)      return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "contextSubscriptions") != NULL)                  return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasecmp(method, "POST") == 0)                                  return DEFAULT_SERVICE_PATH;
+  else if (strcasecmp(method, "PUT") == 0)                                   return DEFAULT_SERVICE_PATH;
+  else if (strcasecmp(method, "DELETE") == 0)                                return DEFAULT_SERVICE_PATH;
+  else if (strcasecmp(method, "GET") == 0)                                   return DEFAULT_SERVICE_PATH_RECURSIVE;
+
+  LM_E(("Runtime Error (cannot found default service path for: (%s, %s))", method, url));
```

Comment:
You are right: if the client send some esoteric HTTP metho such as PATCH, INFO, etc. it is not our fault.